drivers/gpio/gpio_nrfx.c

```diff
@@ -311,13 +298,14 @@ static int gpio_nrfx_port_toggle_bits(struct device *port, u32_t mask)
 }
 
 static int gpio_nrfx_pin_interrupt_configure(struct device *port,
-		unsigned int pin, unsigned int flags)
+		unsigned int pin, enum gpio_int_trigger_mode flags)
 {
 	struct gpio_nrfx_data *data = get_port_data(port);
 	u32_t abs_pin = NRF_GPIO_PIN_MAP(get_port_cfg(port)->port_num, pin);
 
-	if (((flags & GPIO_INT_ENABLE) != 0) &&
-	    ((flags & GPIO_INT_EDGE) != 0) &&
+	if (((flags == GPIO_INT_MODE_EDGE_FALLING) ||
+	     (flags == GPIO_INT_MODE_EDGE_RISING)  ||
+	     (flags == GPIO_INT_MODE_EDGE_BOTH)) &&
```

Comment:
This long if condition would be much simpler if we split `enum gpio_int_trigger_mode` into `enum gpio_int_mode mode, enum gpio_int_trigger trigger`.